### ISSUE TITLE
cpplint.py conflict with Code Style Guide in repository

### ISSUE THREAD

**Issue by iyongliang** [google/styleguide#219]
I found a question about this. the "repository" option can detect .git directory, and the default path is from second level directory. If I have follow files:

```shell
 Example
     - src/
         - hello.h
         - hello.cpp
      - .git/
```

and in hello.h file, 

``` cpp
# ifndef EXAMPLE_SRC_HELLO_H_
# define EXAMPLE_SRC_HELLO_H_
...
# endif  // EXAMPLE_SRC_HELLO_H_
```

cpplint will waring as: header guard has wrong style.

this can fix by:

```c
python cpplint.py --repository=`cd ..;pwd` src/hello.h
```
**But**, when I configure CPPLINT.cfg file, set repository, I found it's a invalid option.
and set 

```c
root=`cd ..;pwd`
```
is not working.

So, can you fix it let root support ``cd ..;pwd`` or add repository to CPPLINT.cfg.

THX!

**Comment by PhilipDeegan** [comment]
I just ran into this

I have a git repo @ ~/ for dot files, and it is detected, I would imagine checking if the CD has a .git repo it should be used first

**Comment by vapier** [comment]
public cpplint development in this repository has been halted (see #837).  please migrate to https://github.com/cpplint/cpplint now.
